sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenMismatchRetryPolicy.java

```diff
@@ -102,11 +107,27 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
                     this.maximumBackoff);
         }
 
+        // For remote region preference ensure that the last retry is long enough (even when exceeding max backoff time)
+        // to consume the entire minRetryTimeInLocalRegion
+        if (sessionRetryOptionsAccessor.getRegionSwitchHint(sessionRetryOptions) ==
+                CosmosRegionSwitchHint.REMOTE_REGION_PREFERRED
+            && attempt >= (this.maxRetryAttemptsInCurrentRegion.get() - 1)) {
```

Comment:
hmm, I am wonder do we really need the maxRetryCount, why the minInRegionRetryTime is not enough( feel it could be hard for the customers to set all these values properly

-      like if maxRetryCount being too large, then the request could stuck in local region much longer than expected.
-  If the maxRetryCount being too low, then we may not hit the minInRegionRetryTime)

Reply:
I think there is a misunderstanding - the if statement will only be entered for the last retry. 

Goal here is to allow avoiding high number of retries to "fill" the minRetryTime (which has to be about the replication latency between regions).

Reply:
Discussed and resolved offline